### PR TITLE
#5 Feature engineering and Model Building codes #7

### PR DESCRIPTION
**Pull request by sumedhkumarprasad**
Feature engineering and Model Building codes #7

### CODE REVIEW

File: src/feature_engg_scripts.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+client, bucket_name = authenticate_s3(which_bucket= "raw_data_bucket_name")
+
+
+df_master = read_from_s3(filename = 'final_master_agg_file',which_bucket= "raw_data_bucket_name")
+df_master.head()
+
+# df_master = pd.read_csv("E:/E2E_DTP_Project/data/final_master_agg_file.csv")
+
+upload_to_s3(df = read_data("E:/E2E_DTP_Project/data/sales.csv"), filename = "sales",which_bucket = "raw_data_bucket_name")
```
Comment (mitulds) [inline src/feature_engg_scripts.py]: why do you have this code here?

File: src/feature_engg_scripts.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
+
+import pandas as pd
+from src.utils import (read_from_s3,read_data,upload_to_s3,authenticate_s3)
+
+client, bucket_name = authenticate_s3(which_bucket= "raw_data_bucket_name")
+
+
+df_master = read_from_s3(filename = 'final_master_agg_file',which_bucket= "raw_data_bucket_name")
+df_master.head()
```
Comment (mitulds) [inline src/feature_engg_scripts.py]: You can read the data inside the process function, don't keep anything outside the functions. Also, can remove line 9

File: src/feature_engg_scripts.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
+
+import pandas as pd
+from src.utils import (read_from_s3,read_data,upload_to_s3,authenticate_s3)
+
+client, bucket_name = authenticate_s3(which_bucket= "raw_data_bucket_name")
```
Comment (mitulds) [inline src/feature_engg_scripts.py]: do you need to authenticate?  since we authenticate already inside upload & read from s3 functions.

**Comment by mitulds** [comment]
@sumedhkumarprasad  you can clean up your scripts. Remove stuff that's not needed anymore, you can store those in other scripts locally.

File: src/feature_engg_scripts.py (hunk 1 above)
Comment (sumedhkumarprasad) [reply src/feature_engg_scripts.py]: Because we need product id, category and unit price columns 
It has connect with line 42 to line 51 in the same code

**Review by mitulds — APPROVED** [review]

**Pull request merged**